.github/workflows/macos_alpha_release.yml

```diff
@@ -0,0 +1,177 @@
+name: macOS - Prepare Alpha Release
+
+defaults:
+  run:
+    working-directory: macOS
+
+on:
+  # To be enabled later
+  # push:
+  #   branches:
+  #     - main
+  workflow_dispatch:
+    inputs:
+      skip-appstore:
+        description: "Skip App Store release and only make a DMG build"
+        default: false
+        type: boolean
+
+jobs:
+
+  calculate-alpha-build-number:
+    name: Calculate Alpha Build Number
+    runs-on: macos-15
+
+    outputs:
+      build-number: ${{ steps.build-number.outputs.next_build_number }}
+
+    steps:
+    - name: Check out the code
+      uses: actions/checkout@v4
+      with:
+        submodules: recursive
+
+    - name: Setup Ruby and dependencies
+      uses: ./.github/actions/setup-ruby
+      with:
+        working-directory: macOS
+        cache-key-prefix: macos-ruby
+
+    - name: Calculate Alpha Build Number
+      id: build-number
+      env:
+        APPLE_API_KEY_BASE64: ${{ secrets.APPLE_API_KEY_BASE64 }}
+        APPLE_API_KEY_ID: ${{ secrets.APPLE_API_KEY_ID }}
+        APPLE_API_KEY_ISSUER: ${{ secrets.APPLE_API_KEY_ISSUER }}
+      run: |
+        bundle exec fastlane run calculate_next_build_number \
```

Comment:
This action calculates the next build number and then uses it in DMG and App Store workflows without committing it to the repo (via `build-number-override`).